plugins/webp-uploads/picture-element.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * Add Picture Element support
+ * Description: Use <picture> element when image has more than one mime type.
+ * Experimental: No
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Potentially wrap an image tag in a picture element.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $image         The image tag.
+ * @param string $context       The context for the image tag.
+ * @param int    $attachment_id The attachment id.
+ *
+ * @return string The new image tag.
+ */
+function webp_uploads_wrap_image_in_picture( string $image, string $context, int $attachment_id ): string {
+	if ( 'the_content' !== $context || ! webp_uploads_picture_element_enabled() ) {
+		return $image;
+	}
+	$image_meta              = wp_get_attachment_metadata( $attachment_id );
+	$original_file_mime_type = get_post_mime_type( $attachment_id );
+	if ( false === $original_file_mime_type || ! isset( $image_meta['sizes'] ) ) {
+		return $image;
+	}
+
+	// Collect all of the sub size image mime types.
+	$mime_type_data = array();
+	foreach ( $image_meta['sizes'] as $size ) {
+		if ( isset( $size['sources'] ) && isset( $size['width'] ) && isset( $size['height'] ) ) {
+			foreach ( $size['sources'] as $mime_type => $data ) {
+				$mime_type_data[ $mime_type ]                         = isset( $mime_type_data[ $mime_type ] ) ? $mime_type_data[ $mime_type ] : array();
```

Comment:
Suggested by PhpStorm:

```suggestion
				$mime_type_data[ $mime_type ]                         = $mime_type_data[ $mime_type ] ?? array();
```